src/Tranche4626.sol

```diff
@@ -0,0 +1,153 @@
+
+// Challenges: 
+// 1. Centrifuge Pools and corresponding Tranches live on Centchain having their liquidity spread across multiple chains. 
+// Latest Tranche values, like share / token price, tranche asset value, total assets... have to be retrieved from Centrifuge chain in order to provide share <-> asset conversions.
+// 2. Pool Epochs: Deposits into and redemptions from Centrifuge Pools are subject to epochs. Deposit and redemption orders are collected during 24H epoch periods
+// and filled during epoch execution following the rules of the underlying pool. Consequently, deposits and redemptions are not instanty possible and have to follow the epoch schedule. 
+// Tranche4626 is extending the EIP4626 standard by 'requestRedeem' & 'requestDeposit' functions, where redeem and deposit orders are submitted to the pools to be included in the execution of the following epoch.
+// After execution users can use the redeem and withdraw functions to get their shares and/or assets from the pools.
+
+// other EIP4626 implementations
+// maple: https://github.com/maple-labs/pool-v2/blob/301f05b4fe5e9202eef988b4c8321310b4e86dc8/contracts/Pool.sol
+// yearn: https://github.com/yearn/yearn-vaults-v3/blob/master/contracts/VaultV3.vy
+
+
+import "./token/restricted.sol";
+
+interface ConnectorLike {
+    function deposit(address _tranche, address _receiver, uint256 _assets) external returns (uint256);
+    function mint(address _tranche, address _receiver, uint256 _shares) external returns (uint256);
+    function maxDeposit(address _user, address _tranche) external returns (uint256);
+    function maxMint(address _user, address _tranche) external returns (uint256);
+    function maxWithdraw(address _user, address _tranche) external returns (uint256);
+    function maxRedeem(address _user, address _tranche) external returns (uint256);
+    function requestRedeem(uint256 _shares, address _receiver) external;
+    function requestDeposit(uint256 _assets, address _receiver) external;
+    
+}
+
+/// @title Tranche4626
+/// @author ilinzweilin
+contract Tranche4626 is RestrictedToken {
+
+    ConnectorLike public connector;
+
+    address public asset; // underlying stable ERC-20 stable currency.
+    uint256 public maxAssetDeposit = 2 ** 256 - 1; // max stable currency deposit into the tranche -> default: no limit.
+
+    uint128 latestPrice; // lates share / token price 
+    uint256 lastPriceUpdate; // timestamp of the latest share / token price update
+   
+    // ERC4626 events
+    event Deposit(address indexed sender, address indexed owner, uint256 assets, uint256 shares);
+    event Withdraw(address indexed sender, address indexed receiver, address indexed owner, uint256 assets, uint256 shares);
+    
+    constructor(address _asset, address _connector, uint8 _decimals, string memory _name, string memory _symbol) super(decimals) {
```

Comment:
I think we should adopt the same pattern as introduced here: https://github.com/centrifuge/connectors/blob/6edcdee1b017d7a0455dfd820245536dedd8c92c/src/token/factory.sol#L26

So only passing `_asset` and `_decimals` on constructor, and filing `_connector`, `_name`, and `_symbol` through a file. That way, we can have deterministic deployments of this contract even if the connector contract changes or the token name/symbol is updated.